### ISSUE TITLE
client.subscribe() keeps creating new connections

### ISSUE THREAD

**Issue by Fishdrowned** [hprose/hprose-html5#7]
I see this in Chome devtool >> Network:
![image](https://user-images.githubusercontent.com/3423411/28448615-dad5c7b6-6e0b-11e7-9bd3-bf4136559d07.png)

Client script:
```html
<script src="../hprose/hprose-html5-2.0.34.js"></script>
<script type="text/javascript">
    !function () {
        var url = 'ws://127.0.0.1:2080/hello',
            subscriber = new hprose.WebSocketClient(url);
        subscriber.subscribe('push', function () {
            console.log(arguments);
        });
    }();
</script>
```

Server script:
```php
use Hprose\Swoole\Server as Server;

$listen = 'ws://0.0.0.0:2080/hello';
$server = new Server($listen);
$server->publish('push');
$server->start();
```

**Comment by Fishdrowned** [comment]
It keep creating new connections about every 30 seconds.

I can see the call stack by putting mouse over the `Initiator`.

The first connection, looks OK, it was initiated by `subscribe()`:
![image](https://user-images.githubusercontent.com/3423411/28449061-4d2323c4-6e0f-11e7-865c-88d25cb2ac78.png)

The second connection, no good:
![image](https://user-images.githubusercontent.com/3423411/28449189-87d2aca0-6e10-11e7-9f6a-7a25dd89657c.png)

The third connection, wow, look at the scroll bar, it gets longer:
![image](https://user-images.githubusercontent.com/3423411/28449287-222a7f4e-6e11-11e7-91b2-6030771f47c8.png)

**Comment by andot** [comment]
because the client default timeout is 30s, and the server default timeout is 120s. you can the the client timeout more than 120s to fixed problem.

**Comment by Fishdrowned** [comment]
Thank you for your reply, I've extended the client timeout to 1 day, it stops creating new connections.

But look at the 1st screenshot, it indicates that the old connections are not closed (status = `101` and time = `Pending` means the connection is still alive), but new connections are created.

PS. The term [`heartbeat`](https://github.com/hprose/hprose-php/wiki/10.-%E6%8E%A8%E9%80%81%E6%9C%8D%E5%8A%A1#heartbeat-%E5%B1%9E%E6%80%A7) is misleading, I expected it to send some `ping` frames to the client to keep connection open.

**Comment by andot** [comment]
When timeout on client, the connection will not to be closed, because maybe some other requests on this connection too.

**Comment by Fishdrowned** [comment]
The problem is, by default settings, `client.subscribe()` will keep creating connections as time grows, keep them open, not reusable, never closed, just like memory leak.

I guess this should be a architecture issue, but I don't have enough time to investigate and prove this.

Sets the client timeout to 1 day or some big number is just a workaround.

I implemented a manual but **real** heartbeat in my application, by pushing an empty string to the subscribers every minute.

**Comment by andot** [comment]
Yes, you are right. I will  try to fix this problem in  next version.
